### PR TITLE
Add support for using sccache on Unix

### PR DESCRIPTION
**Pull request by agocke**
sccache (https://github.com/mozilla/sccache) is a compiler caching tool that supports local or remote storage. It's useful in /runtime for caching the results of the C++ components of the build.

This PR adds support for dotnet/runtime by recognizing a $USE_SCCACHE=true environment variable, which then instructs CMake to look for a the sccache binary in the PATH.

This support is still experimental in dotnet/runtime and only on Linux/Mac. It's not confirmed that caching is completely safe for the build.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
It looks like sccache also supports MSVC according to their readme, so it might be worth adding support into the Windows build scripts as well.

I saw that [ccache added support for MSVC](https://ccache.dev/releasenotes.html) recently. Would it be worth trying to provide ccache as an option as well?

**Review by jkoritzinsky — APPROVED** [review]

File: src/coreclr/build-runtime.sh [hunk 1]
```diff
@@ -165,6 +165,10 @@ if [[ "$__TargetArch" != "$__HostArch" ]]; then
     __CMakeArgs="-DCLR_CMAKE_TARGET_ARCH=$__TargetArch $__CMakeArgs"
 fi
 
+if [[ "$USE_SCCACHE" == "true" ]]; then
+    __CMakeArgs="-DCMAKE_CXX_COMPILER_LAUNCHER=sccache"
```
Comment (jkoritzinsky) [inline src/coreclr/build-runtime.sh:169]:
```suggestion
    __CMakeArgs="-DCMAKE_C_COMPILER_LAUNCHER=sccache -DCMAKE_CXX_COMPILER_LAUNCHER=sccache $__CMakeArgs"
```

Comment (agocke) [reply src/coreclr/build-runtime.sh:169]: oops!

**Comment by agocke** [comment]
Failures are all unrelated to this change.

**Pull request merged**